frontend-wasm2/src/component/build_ir.rs

```diff
@@ -32,27 +29,24 @@ pub fn translate_component(
     config: &WasmTranslationConfig,
     context: Rc<Context>,
 ) -> WasmResult<midenc_hir2::dialects::builtin::ComponentRef> {
-    let (mut component_types_builder, parsed_root_component) =
+    let (mut component_types_builder, mut parsed_root_component) =
         parse(config, wasm, context.session())?;
-    let dialect = context.get_or_register_dialect::<BuiltinDialect>();
-    dialect.expect_registered_name::<midenc_hir2::dialects::builtin::Component>();
-    // context.get_or_register_dialect::<HirDialect>();
     // Extract component name from exported component instance
     let id = {
-        let instance = parsed_root_component
+        let instance: String = parsed_root_component
```

Comment:
*Nit:* This type annotation can be elided no? The call to `.to_string()` disambiguates it, and the signature of `parse_component_id` requires a `String` anyway.

Doesn't really matter too much, but Rust already can be a bit noisy at times with types.

Reply:
Yep. Thanks. Done.